doc/source/rllib/rllib-offline.rst

```diff
@@ -8,217 +8,662 @@ Working With Offline Data
+
+.. testsetup:: 
+    
+    # Define a shared variable to store the path to the
+    # best checkpoint.
+    best_checkpoint = None
+
+    # Define a shared variable to store the path to the
+    # recorded data.
+    data_path = None
+
+    # Define another shared variable to store the path to
+    # the tabular recording data.
+    tabular_data_path = None
+
+.. testcode::
+
+    from ray.rllib.algorithms.ppo import PPOConfig
+    from ray.rllib.core.rl_module.default_model_config import DefaultModelConfig
+    from ray.rllib.utils.metrics import (
+        ENV_RUNNER_RESULTS,
+        EVALUATION_RESULTS,
+        EPISODE_RETURN_MEAN,
+    )
+    from ray import train, tune
 
-RLlib represents trajectory sequences (i.e., ``(s, a, r, s', ...)`` tuples) with `SampleBatch <https://github.com/ray-project/ray/blob/master/rllib/policy/sample_batch.py>`__ objects. Using a batch format enables efficient encoding and compression of experiences. During online training, RLlib uses `policy evaluation <rllib-concepts.html#policy-evaluation>`__ actors to generate batches of experiences in parallel using the current policy. RLlib also uses this same batch format for reading and writing experiences to offline storage.
+    # Configure the PPO algorithm.
+    config = (
+        PPOConfig()
+        .environment("CartPole-v1")
+        # Enable the new API stack.
+        .api_stack(
+            enable_rl_module_and_learner=True,
+            enable_env_runner_and_connector_v2=True,
+        )
+        .training(
+            lr=0.0003,
+            # Run 6 SGD minibatch iterations on a batch.
+            num_epochs=6,
+            # Weigh the value function loss smaller than 
+            # the policy loss.
+            vf_loss_coeff=0.01,
+        )
+        .rl_module(
+            model_config=DefaultModelConfig(
+                fcnet_hiddens=[32],
+                fcnet_activation="linear",
+                # Share encoder layers between value network
+                # and policy.
+                vf_share_layers=True,
+            ),
+        )
+    )
 
-Example: Training on previously saved experiences
--------------------------------------------------
+    # Define the metric to use for stopping.
+    metric = f"{EVALUATION_RESULTS}/{ENV_RUNNER_RESULTS}/{EPISODE_RETURN_MEAN}"
 
-.. note::
+    # Define the Tuner.
+    tuner = tune.Tuner(
+        "PPO",
+        param_space=config,
+        run_config=train.RunConfig(
+            stop={
+                metric: 450.0,
+            },
+            name="docs_rllib_offline_pretrain_ppo",
+            verbose=2,
+            checkpoint_config=train.CheckpointConfig(
+                checkpoint_frequency=1,
+                checkpoint_at_end=True,
+            ),
+        ),
+    )
+    analysis = tuner.fit()
 
-    For custom models and enviroments, you'll need to use the `Python API <rllib-training.html#basic-python-api>`__.
+    best_checkpoint = (
```

Comment:
nit: add some comments here about what we are going to do with this checkpoint next ...

Reply:
Yeah! Good point!